internal/resource/tree.go

```diff
@@ -137,6 +143,7 @@ func (t *tree) Get(key Ref) (res *Resource, visible bool, found bool) {
 	if !ok {
 		return nil, false, false
 	}
+	//debug logging on what we're blocked on might help future issues.
```

Comment:
we definitely need something to help find deadlocks. All of the obvious places to add logging would be extremely noisy - would it have been useful in your case to expose the `tree` as json on the loopback interface or something?

I'm thinking a debugging workflow like port-forward to eno-controller and `curl localhost:8000/tree/<compositionname>`